### PR TITLE
Solidity 0.5 update

### LINKED ISSUE #1078: Solidity 0.5 support
**Describe the solution you'd like**
We would like to upgrade to the latest version of Solidity.  This is not critical, but a nice to have improvement.

**Additional context**
Changing from Solidity 0.4 to 0.5 includes breaking changes.  Many .sol files cannot be written in such a way as to support both versions (some can, like ZOS's `Initializable`).

At the moment there is one blocker preventing us from completing the upgrade: https://github.com/OpenZeppelin/openzeppelin-eth specifically the `Ownable` implementation (the other import, `ERC165`, is available in openzeppelin-solidity).

### PR DESCRIPTION
**Pull request by hardlydifficult**
# Description

Upgrading to the latest Solidity version.  This must be done along side Truffle and OpenZeppelin updates, which causes some breaks to our test code.  

One impact of note is the gas cost to `createLock` increased a bit.

# Issues

Fixes #1078
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by hardlydifficult** [comment]
This requires testing the unlock-app.  On first attempt, I got an `out of gas` error.  Will investigate

File: smart-contracts/test/Lock/erc721/safeTransferFrom.js [hunk 1]
```diff
@@ -37,53 +34,24 @@ contract('Lock ERC721', accounts => {
...
-            },
-            {
-              type: 'bytes',
-              name: 'data'
-            }
-          ]
-        },
-        [sender, receiver, Web3Utils.toHex(ID), Web3Utils.toHex('Julien')]
+      await locks['FIRST'].methods['safeTransferFrom(address,address,uint256,bytes)'](
+        accounts[7], accounts[6], Web3Utils.toHex(ID), Web3Utils.toHex('Julien'), { from: accounts[7] }
```
Comment (nfurfaro) [inline smart-contracts/test/Lock/erc721/safeTransferFrom.js]: Nice. I was hoping we could lose the `encodeFunctionCall` with this update.

**Comment by hardlydifficult** [comment]
Researching the out of gas error I saw yesterday, it appears to be because I did not `nuke` first.  Running the `unlock-app` one more time from a fresh pull to be sure.

File: smart-contracts/test/Lock/erc721/safeTransferFrom.js (hunk 1 above)
Comment (hardlydifficult) [reply smart-contracts/test/Lock/erc721/safeTransferFrom.js]: Yes, this is much nicer.  I would have saved it for a separate PR though... however the original method stopped working.  I'm unsure why

**Pull request merged**